### PR TITLE
do not preload hidden files and files in hidden folders

### LINKED ISSUE #682: Do not preload hidden files and files in hidden folders
Hi!

First of all thank you for such a great project :)

We use GreenMail in a local k8s cluster (minikube) to check if everything works as expected. It is a common practice for such a scenario to mount emails for preloading into container via configMap volume mount. But k8s adds some extra hidden folders, as a result we obtain exceptions like `com.icegreen.greenmail.store.FolderException: Invalid mailbox name ..2024_02_09_11_29_23.577733621`. My suggestion is to ignore hidden files and files which are in hidden folders during emails preload.

### PR DESCRIPTION
**Pull request by An1s9n**
Resolves #682

### CODE REVIEW

File: greenmail-core/src/test/resources/preload/foo-bar@localhost/.keepme
Comment (marcelmay) [inline greenmail-core/src/test/resources/preload/foo-bar@localhost/.keepme:1]: Required for test to succeed, please do not delete this

File: greenmail-core/src/test/resources/preload/drafts@localhost/Drafts/.keepme
Comment (marcelmay) [inline greenmail-core/src/test/resources/preload/drafts@localhost/Drafts/.keepme:1]: Required for test to succeed, please do not delete this

**Review by marcelmay — CHANGES_REQUESTED** [review]
Please do not remove the mentioned foo-bar and drafts folders. These make the test fail.

Otherwise looks good, thanks alot!

File: greenmail-core/src/test/resources/preload/foo-bar@localhost/.keepme
Comment (An1s9n) [reply greenmail-core/src/test/resources/preload/foo-bar@localhost/.keepme:1]: Yeah, my bad. Of course you are right, git does not track empty folders. Fixed that.

File: greenmail-core/src/test/resources/preload/drafts@localhost/Drafts/.keepme
Comment (An1s9n) [reply greenmail-core/src/test/resources/preload/drafts@localhost/Drafts/.keepme:1]: Yeah, my bad. Of course you are right, git does not track empty folders. Fixed that.

**Pull request merged**